pkg/agent/cniserver/interface_configuration_linux.go

```diff
@@ -92,49 +93,169 @@ func (ic *ifConfigurator) moveIfToNetns(ifname string, netns ns.NetNS) error {
 	return nil
 }
 
-func (ic *ifConfigurator) moveVFtoContainerNS(vfNetDevice string, containerID string, containerNetNS string, containerIfaceName string, mtu int, result *current.Result) error {
+func (ic *ifConfigurator) configureVFLinkAndIPAM(link netlink.Link, containerID string, containerIfaceName string, netnsPath string, mtu int, result *current.Result) error {
 	hostIface := result.Interfaces[0]
 	containerIface := result.Interfaces[1]
 
+	err := ic.netlink.LinkSetMTU(link, mtu)
+	if err != nil {
+		return fmt.Errorf("failed to set MTU for VF netdevice %s: %w", containerIfaceName, err)
+	}
+	err = ic.netlink.LinkSetUp(link)
+	if err != nil {
+		return fmt.Errorf("failed to set link up to VF netdevice %s: %w", containerIfaceName, err)
+	}
+	containerIface.Name = containerIfaceName
+	containerIface.Mac = link.Attrs().HardwareAddr.String()
+	containerIface.Sandbox = netnsPath
+	klog.V(2).Infof("Configuring IP address for container %s, hostIface: %+v, containerIface: %+v", containerID, hostIface, containerIface)
```

Comment:
please switch to structured logging for every new log or every log you "touch" as part of a PR